docs/source/design/log_levels.rst

```diff
@@ -0,0 +1,32 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+**********
+Log levels
+**********
+
+Log messages are emitted by a wide range of operations within DataLad. They are
+categorized into distinct levels. While some levels have self-explanatory
+descriptions (e.g. ``warning``, ``error``), others are less specific (e.g.
+``info``, ``debug``).
+
+Common principles
+=================
+
+Parenthical log message use the same level
+  When log messages are used to indicate the start and end of an operation,
+  both start and end message use the same log-level.
+
+Use cases
+=========
+
+Command execution
+-----------------
+
+For the non-async Runner and its protocols the following log levels are used:
+
+- High-level execution -> ``debug``
+- Process start/finish -> ``8``
+- Threading and IO -> ``5``
```

Comment:
May be here add a section on related config/envvar (may be some envvars no longer in use) settings ?

<details>
<summary>what I see for config + envvars in CONTRIBUTING.md</summary> 

```shell
(git)lena:~datalad/datalad-master[master]git
$> git grep 'datalad\.log' datalad/interface/common_cfg.py
datalad/interface/common_cfg.py:    'datalad.log.level': {
datalad/interface/common_cfg.py:    'datalad.log.result-level': {
datalad/interface/common_cfg.py:    'datalad.log.name': {
datalad/interface/common_cfg.py:    'datalad.log.names': {
datalad/interface/common_cfg.py:    'datalad.log.namesre': {
datalad/interface/common_cfg.py:    'datalad.log.outputs': {
datalad/interface/common_cfg.py:    'datalad.log.timestamp': {
datalad/interface/common_cfg.py:    'datalad.log.traceback': {
datalad/interface/common_cfg.py:    'datalad.log.exc': {

$> grep LOG CONTRIBUT*                                    
CONTRIBUTING.md:- *DATALAD_LOG_LEVEL*:
CONTRIBUTING.md:- *DATALAD_LOG_NAME*:
CONTRIBUTING.md:- *DATALAD_LOG_OUTPUTS*:
CONTRIBUTING.md:- *DATALAD_LOG_PID*
CONTRIBUTING.md:- *DATALAD_LOG_TARGET*
CONTRIBUTING.md:- *DATALAD_LOG_TIMESTAMP*:
CONTRIBUTING.md:- *DATALAD_LOG_TRACEBACK*:
CONTRIBUTING.md:- *DATALAD_LOG_VMEM*:

```
</details>

Reply:
(see main comment)